packages/synapse-react-client/src/synapse-queries/entity/useGetEntityHeaders.ts

```diff
@@ -13,23 +14,76 @@ export function useGetEntityHeaders(
   options?: UseQueryOptions<PaginatedResults<EntityHeader>, SynapseClientError>,
 ) {
   const { accessToken, keyFactory } = useSynapseContext()
+  const queryClient = useQueryClient()
+  const queryFn = async () => {
+    const response = await SynapseClient.getEntityHeaders(
+      references,
+      accessToken,
+    )
+
+    // Update the cache with the individual header, in case this was a batch
+    response.results.forEach(header => {
+      const requestedItem = references.find(ref => {
+        if (ref.targetVersionNumber) {
+          return (
+            ref.targetId === header.id &&
+            ref.targetVersionNumber === header.versionNumber
+          )
+        } else {
+          return ref.targetId === header.id
+        }
+      })
+      if (requestedItem) {
+        queryClient.setQueryData<PaginatedResults<EntityHeader>>(
+          keyFactory.getEntityHeadersQueryKey([requestedItem]),
+          { results: [header], totalNumberOfResults: 1 },
+        )
+
+        queryClient.setQueryData(
+          keyFactory.getEntityHeaderQueryKey(
+            requestedItem.targetId,
+            requestedItem.targetVersionNumber,
+          ),
+          header,
+        )
+
+        // If the request item had a version and was the latest version, we can also hydrate the cache
+        // for the 'unversioned' request
+        if (requestedItem.targetVersionNumber && header.isLatestVersion) {
+          const requestWithoutVersionNumber: Reference = { targetId: header.id }
+          queryClient.setQueryData<PaginatedResults<EntityHeader>>(
+            keyFactory.getEntityHeadersQueryKey([requestWithoutVersionNumber]),
+            { results: [header], totalNumberOfResults: 1 },
+          )
+
+          queryClient.setQueryData(
+            keyFactory.getEntityHeaderQueryKey(requestedItem.targetId),
+            header,
+          )
+        }
+      }
```

Comment:
This is how we're going to pre-load the react-query cache with all of the entity headers